lnwallet/channel.go

```diff
@@ -6039,6 +6070,14 @@ func (lc *LightningChannel) availableCommitmentBalance(view *htlcView) (
 		}
 
 		ourBalance -= htlcCommitFee
+
+	// If we're not the initiator, we must check whether the remote has
+	// enough balance to pay for the fee of our HTLC. If the they cannot
+	// pay the fee if we add another non-dust HTLC, set our available
+	// balance just below the non-dust amount, to avoid attempting HTLCs
+	// larger than this size.
+	case theirBalance < htlcCommitFee && ourBalance >= nonDustHtlcAmt:
```

Comment:
I think the readability of this function isn't optimal. There are two relatively independent cases: initiator and non-initiator. But several variables are set in the parent scope, making it less clear what is used where. Maybe converting this `switch` to a classic `if initiator then .. else ..` and defining the specific vars in the if blocks is better.

Reply:
Is this better? https://github.com/lightningnetwork/lnd/pull/3691/commits/ae2a41b3464133cd80d38021c48c172796f4c3a1

Reply:
Yes